### PR TITLE
Support teams engine snapshots

### LINKED ISSUE #8917: Support snapshot for teams

### PR DESCRIPTION
**Pull request by ValentinTrinque**
Close #8917

### CODE REVIEW

File: core/teams/snapshot.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+func (e *SnapshotEngine) loadTeams(p *types.Payload, teams []*snapshotpb.Team) error {
+	e.Engine.loadTeamsFromSnapshot(teams)
+
+	serialisedTeams, err := proto.Marshal(p.IntoProto())
+	if err != nil {
+		return fmt.Errorf("could not serialize teams payload: %w", err)
+	}
+
+	e.serialisedTeams = serialisedTeams
```
Comment (jeremyletang) [inline core/teams/snapshot.go:170]: any reason we keep all the serialised state around? I could not see where we reuse it ever?

File: core/teams/snapshot.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+}
+
+func (e *SnapshotEngine) serialiseTeams() error {
+	teams := e.Engine.teams
+	teamsSnapshot := make([]*snapshotpb.Team, 0, len(teams))
+	for _, team := range teams {
+		refereesSnapshot := make([]string, 0, len(team.Referees))
+		for _, referee := range team.Referees {
+			refereesSnapshot = append(refereesSnapshot, string(referee))
+		}
```
[the remark this answers is not in the document]
Comment (ValentinTrinque) [reply core/teams/snapshot.go:92]: sorted by time of arrival, mostly

File: core/teams/snapshot.go (hunk 1 above)
Comment (ValentinTrinque) [reply core/teams/snapshot.go:170]: True, I blindly reproduced what we do everywhere. I think this things should disappear. Some project-wide cleanup should be done regarding this. This is probably a left-over from the initial snapshot implementation

File: core/teams/snapshot.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ValentinTrinque) [reply core/teams/snapshot.go:92]: Do you want to have it sorted alpha-num ?

File: core/teams/snapshot.go (hunk 1 above)
Comment (ValentinTrinque) [reply core/teams/snapshot.go]: Removed.

**Review by candida-d — DISMISSED** [review]

**Review by wwestgarth — DISMISSED** [review]

**Review by gordsport — DISMISSED** [review]
LGTM

**Pull request merged**